xbmc/interfaces/builtins/PlayerBuiltins.cpp

```diff
@@ -39,6 +41,45 @@
 #include "Autorun.h"
 #endif
 
+namespace
+{
+bool IsMediasourceRestricted(const CFileItem& item)
+{
+  const auto settingsComponent = CServiceBroker::GetSettingsComponent();
+  if (!settingsComponent)
+  {
+    CLog::LogF(LOGERROR, "GetSettingsComponent() returned nullptr.");
```

Comment:
This is just an idle comment, no need for action.
However i wonder if we should "standardise" a log string. From memory @garbear knew all the ins and outs of __FUNCTION__ and its variants on different platforms.

Something like "CCLASS::FUNC - message". Just so its more obvious exactly where a log message is being generated.

Reply:
yes, coolest would be this https://en.cppreference.com/w/cpp/utility/source_location. bet until then we'll have to grep the codebase for interesting messages without __ func __ :/

Reply:
`source_location` looks nice but doesn't handle the class name. IIRC on windows `__FUNCTION__` includes the class name. Also IIRC not all compilers have a way to "detect" which class a function is in.

Reply:
Here's my comment: https://github.com/xbmc/xbmc/pull/20254#discussion_r722728647